src/db/dal/user.ts

```diff
@@ -114,9 +276,60 @@ export const completeRegistration = async (
     return results[1][0];
 };
 
-export const deleteUser = async (keycloak_id: string): Promise<boolean> => {
-    const deletedCount = await UserModel.destroy({
-        where: { keycloak_id },
+export const updateRolesAndDataUsages = async (): Promise<void> => {
+    const results = await UserModel.findAll();
+
+    results.map(async (user) => {
+        await UserModel.update(
+            {
+                ...user,
+                updated_date: new Date(),
+                roles: replaceRoles(user.roles || []),
+                portal_usages: replacePortalUsages(user.portal_usages || []),
+            },
+            {
+                where: {
+                    keycloak_id: user.keycloak_id,
+                },
+                returning: true,
+            },
+        );
     });
-    return !!deletedCount;
 };
+
+const replaceRoles = (roles: string[]): string[] =>
```

Comment:
I think we should not have to "translate" for the portal - client should sent the correct value to this service.

Reply:
C'est pour une migration one-shot pour les roles déjà insérés avec les mauvaises valeurs